### PR TITLE
Add CreatedEvent.createdAt in Java bindings

### PR DESCRIPTION
**Pull request by tudor-da**
Java bindings adaptation after https://github.com/digital-asset/daml/pull/17726

### CODE REVIEW

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java [hunk 1]
```diff
@@ -95,7 +103,8 @@ public CreatedEvent(
         agreementText,
         contractKey,
         signatories,
-        observers);
+        observers,
+        Instant.EPOCH);
```
Comment (cocreature) [inline language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java:107]: hm feels a bit fishy but better than making it an optional. Let's stick with epoch

**Review by cocreature — APPROVED** [review]
Nice thank you!

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java (hunk 1 above)
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java:107]: yeah, I pondered on it and decided that Optional is messy. Kept it clean and added a comment over the getter

**Review by mziolekda — APPROVED** [review]

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java [hunk 2]
```diff
@@ -39,6 +40,11 @@ public final class CreatedEvent implements Event, TreeEvent {
 
   private final @NonNull ByteString createdEventBlob;
 
+  // Note that we can't use a `com.daml.ledger.javaapi.data.Timestamp` here because
+  // it only supports milliseconds-precision and we require lossless conversions through
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java]: Generally I agree, but going with `data.Timestamp` introduces an inconsistency with the representation of `Transaction.effectiveAt` (which also uses `Instant`). Furthermore, `CreatedEvent` is read-only from a client's POV so no issue in losing precision on construction. For consistency and Java-friendly API-compat, I'd keep it like this

[the remark this answers is not in the document]
Comment (raphael-speyer-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java]: Fair enough on the use of `Instant`. 

The code comment suggestion was separate, though. The granularity of `data.Timestamp` is microseconds, not milliseconds.

**Review by raphael-speyer-da — APPROVED** [review]
LGTM pending `s/milli/micro/` doc change.

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java]: I've accepted the code-change suggestion, thanks!

**Pull request merged**